sw/airborne/modules/sensors/airspeed_uavcan.c

```diff
@@ -0,0 +1,106 @@
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+/** @file modules/sensors/airspeed_uavcan.c
+ * Airspeed sensor on the uavcan bus
+ */
+
+#include "airspeed_uavcan.h"
+#include "uavcan/uavcan.h"
+#include "core/abi.h"
+#include "filters/low_pass_filter.h"
+
+#include "pprzlink/messages.h"
+#include "modules/datalink/downlink.h"
+
+#if PERIODIC_TELEMETRY
+#include "modules/datalink/telemetry.h"
+#endif
+
+#ifdef USE_AIRSPEED_LOWPASS_FILTER
+static Butterworth2LowPass airspeed_filter;
+#endif
+static uavcan_event airspeed_uavcan_ev;
+
+/* uavcan EQUIPMENT_ESC_STATUS message definition */
+#define UAVCAN_EQUIPMENT_AIR_DATA_RAWAIRDATA_ID            1027
+#define UAVCAN_EQUIPMENT_AIR_DATA_RAWAIRDATA_SIGNATURE     (0xC77DF38BA122F5DAULL)
+#define UAVCAN_EQUIPMENT_AIR_DATA_RAWAIRDATA_MAX_SIZE      ((397 + 7)/8)
+
+struct airspeed_uavcan_s airspeed_uavcan;
+
+static void airspeed_uavcan_downlink(struct transport_tx *trans, struct link_device *dev)
+{
+  pprz_msg_send_AIRSPEED_UAVCAN(trans,dev,AC_ID,
+                                &airspeed_uavcan.diff_p,
+                                &airspeed_uavcan.temperature);
+}
+
+static void airspeed_uavcan_cb(struct uavcan_iface_t *iface __attribute__((unused)), CanardRxTransfer *transfer) {
+  uint16_t tmp_float = 0;
+  float diff_p;
+
+  /* Decode the message */
+  //canardDecodeScalar(transfer, (uint32_t)0, 8, false, (void*)&dest->flags);
+  //canardDecodeScalar(transfer, (uint32_t)8, 32, false, (void*)&static_p);
+  canardDecodeScalar(transfer, (uint32_t)40, 32, false, (void*)&diff_p);
+  //canardDecodeScalar(transfer, (uint32_t)72, 16, false, (void*)&tmp_float);
+  //float static_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+  //canardDecodeScalar(transfer, (uint32_t)88, 16, false, (void*)&tmp_float);
+  //float diff_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+  canardDecodeScalar(transfer, (uint32_t)104, 16, false, (void*)&tmp_float);
+  float static_air_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+  //canardDecodeScalar(transfer, (uint32_t)120, 16, false, (void*)&tmp_float);
+  //float pitot_temp = canardConvertFloat16ToNativeFloat(tmp_float);
+
+  if(!isnan(diff_p)) {
+#ifdef USE_AIRSPEED_LOWPASS_FILTER
+    float diff_p_filt = update_butterworth_2_low_pass(&airspeed_filter, diff_p);
+    airspeed_uavcan.diff_p = diff_p;
+    //AbiSendMsgBARO_DIFF(UAVCAN_SENDER_ID, diff_p_filt);
+#else
+    //AbiSendMsgBARO_DIFF(UAVCAN_SENDER_ID, diff_p);
+    airspeed_uavcan.diff_p = diff_p;
+#endif
+  }
+
+  if(!isnan(static_air_temp))
+    //AbiSendMsgTEMPERATURE(UAVCAN_SENDER_ID, static_air_temp);
+    airspeed_uavcan.temperature = static_air_temp;
+}
+
+void airspeed_uavcan_init(void)
+{
+  // Setup low pass filter with time constant and 100Hz sampling freq
+#ifdef USE_AIRSPEED_LOWPASS_FILTER
+  init_butterworth_2_low_pass(&airspeed_filter, MS45XX_LOWPASS_TAU, MS45XX_I2C_PERIODIC_PERIOD, 0);
```

Comment:
Change the `TAU` and `PERIOD`, I guess the `PERIOD` should be a simple define and add both to the doc of the module.

Reply:
done